lib/services/local/users.go

```diff
@@ -1215,6 +1215,62 @@ func (s *IdentityService) GetSAMLAuthRequest(id string) (*services.SAMLAuthReque
 	return &req, nil
 }
 
+// CreateSSODiagnosticInfo creates new SAML diagnostic info record.
+func (s *IdentityService) CreateSSODiagnosticInfo(ctx context.Context, authKind string, authRequestID string, entry types.SSODiagnosticInfo) error {
+	if authRequestID == "" {
+		return trace.BadParameter("missing parameter authRequestID")
+	}
+
+	switch authKind {
+	case types.KindSAML, types.KindGithub, types.KindOIDC:
+		// nothing to do
+	default:
+		return trace.BadParameter("unsupported authKind %q", authKind)
+	}
+
+	jsonValue, err := json.Marshal(entry)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	item := backend.Item{
+		Key:     backend.Key(webPrefix, connectorsPrefix, authKind, requestsTracePrefix, authRequestID),
+		Value:   jsonValue,
+		Expires: backend.Expiry(s.Clock(), time.Minute*15),
+	}
+	_, err = s.Create(ctx, item)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// GetSSODiagnosticInfo returns SSO diagnostic info records.
+func (s *IdentityService) GetSSODiagnosticInfo(ctx context.Context, authKind string, authRequestID string) (*types.SSODiagnosticInfo, error) {
+	if authRequestID == "" {
+		return nil, trace.BadParameter("missing parameter id")
+	}
+
+	switch authKind {
+	case types.KindSAML, types.KindGithub, types.KindOIDC:
+		// nothing to do
```

Comment:
```suggestion
```

Reply:
As in the other place: the comment is there to indicate this is an explicit choice to do nothing. A seasoned reader will recognize a pattern, but a comment will make it unambiguous for everyone, including less experienced readers.